src/composables/storageRef.ts

```diff
@@ -56,28 +59,28 @@ export function useStorageRef<T = string | object | any[]>(
   const state = ref(initialState) as Ref<T>; // https://github.com/vuejs/core/issues/2136/
   const storage = (enableSecureStorage && IS_MOBILE_APP) ? SecureMobileStorage : WalletStorage;
 
-  async function setLocalState(val: T | null) {
+  async function setLocalState(val: TSerialized | null) {
     if (val !== null) {
       watcherDisabled = true;
-      state.value = (serializer?.read) ? await serializer.read(val) : val;
+      state.value = await serializer.read(val);
       setTimeout(() => { watcherDisabled = false; }, 0);
     }
   }
 
   async function setStorageState(val: T | null) {
-    storage.set(storageKey, (val && serializer?.write) ? await serializer.write(val) : val);
+    storage.set(storageKey, val ? await serializer.write(val) : val);
   }
 
   // Restore state and run watchers
   (async () => {
-    let restoredValue = await storage.get<T | null>(storageKey);
+    let restoredValue = await storage.get<TSerialized | null>(storageKey);
     if (migrations?.length) {
-      restoredValue = await asyncPipe<T | null>(migrations)(restoredValue!);
+      restoredValue = await asyncPipe<TSerialized | null>(migrations)(restoredValue!);
       if (restoredValue !== null) {
-        await setStorageState(restoredValue);
```

Comment:
I think `setStorageState` here was used in case we would want to introduce serializer to an existing (stored) value,